Observer/OfflinePaymentObserver.php

```diff
@@ -39,32 +59,40 @@ public function execute(Observer $observer)
     {
         $order   = $observer->getEvent()->getOrder();
         $payment = $order->getPayment();
-
-        if ($payment->getAdditionalData('payment_type') !== 'bill_payment_tesco_lotus') {
-            return $this;
+        $emailData     = new \Magento\Framework\DataObject();
+        switch($payment->getAdditionalInformation('payment_type')) {
+            case 'bill_payment_tesco_lotus':
+                // make sure timezone is Thailand.
+                date_default_timezone_set("Asia/Bangkok");
+                $barcodeHtml   = $this->_helper->convertTescoSVGCodeToHTML($payment->getAdditionalInformation('barcode'));
+                $emailTemplate = 'send_email_tesco_template';
+                // get DateTime deadline that is in next 24 hours.
+                $validUntil    = date("d-m-Y H:i:s" , time() + 24 * 60 * 60);
```

Comment:
Any reason not to use `expires_at` for Tesco Lotus Bill Payment as well?

See charge response here: https://www.omise.co/bill-payment#creating-a-charge

Reply:
It was like this since before, I think better if we use it.

Reply:
Done